src/InstagramClone.Api/Database/InstagramCloneDbContext.cs

```diff
@@ -0,0 +1,41 @@
+﻿using InstagramClone.Api.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace InstagramClone.Api.Database
+{
+    public class InstagramCloneDbContext : DbContext
+    {
+        public InstagramCloneDbContext(): base()
+        {
+
+        }
+
+       
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<User>().HasKey(u => u.Id);
+            modelBuilder.Entity<Post>().HasKey(p => p.Id);
+            modelBuilder.Entity<PostReaction>().HasKey(pr => pr.Id);
+
+            // composite Key
+            modelBuilder.Entity<UserFollower>().HasKey(userfollower => new {userfollower.FollowedUserId, userfollower.FollowerId});
+           
+
+            // one to many relationship
+            modelBuilder.Entity<Post>().HasOne<User>(post => post.Author).WithMany( user => user.Posts);
+            modelBuilder.Entity<PostReaction>().HasOne<Post>(reaction => reaction.Post).WithMany(posts => posts.PostReactions);
+            modelBuilder.Entity<UserFollower>().HasOne<User>(follower => follower.FollowedUser).WithMany(u => u.Users);
```

Comment:
This relationship isn't what you think it is

What we have is a Many to Many Relationship ...the only Many to Many Relationship in the data model...and it's from User to User. It's called a Self-referencing M2M relationship so sometimes it can be difficult to configure.

So what we have is

1 `User` => (m) `UserFollower` + (m) `UserFollower` (m) => 1 `User` 

which gives us (m) `User` => (m) `User` (many 2 many)

You should read this and tell me if you see some ideas 
https://www.entityframeworktutorial.net/efcore/configure-many-to-many-relationship-in-ef-core.aspx

Reply:
Also watch this video afterwards https://www.youtube.com/watch?v=Qh2hgIc90y0